### PR TITLE
multiple fixes to LDAP EAP-TTLS PAP

### LINKED ISSUE #6571: LDAP TTLS PAP broken
**Describe the bug**
Looks like the LDAP TTLS PAP implementation is broken since our 10.2 release
The field was renamed to `ldap_source_ttls_pap` in the realms but wasn't adjusted in `lib/pf/services/manager/radiusd_child.pm` so it never does anything and actually prevents radiusd from starting when enabled

**To Reproduce**
1. Configure an LDAP ttls PAP source in the NULL realm
2. Restart radiusd

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
A bunch of fixes for EAP TTLS PAP (see news below)

# Impacts
All of EAP-TTLS PAP

# Issue
fixes #6571
starts addressing #6570 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* Match the realm more strictly when its not a regex in EAP-TTLS PAP
* Populate the LDAP config for enabled LDAP EAP-TTLS PAP realms
* Only call oauth2 in authorize for the realms that have an Azure AD EAP-TTLS PAP configuration
* Use source username in LDAP module for EAP-TTLS PAP instead of always using sAMAccoutName
* Support LDAP certificate client auth for LDAP EAP-TTLS PAP authentication
* Allow to use Google Workspace LDAP sources in EAP-TTLS PAP authentication

### CODE REVIEW

File: lib/pf/services/manager/radiusd_child.pm [hunk 1]
```diff
@@ -731,7 +731,7 @@ sub generate_radiusd_ldap {
     foreach my $ldap (keys %ConfigAuthenticationLdap) {
         my $active = $FALSE;
         foreach my $realm ( @pf::config::ConfigOrderedRealm ) {
-            if (defined($pf::config::ConfigRealm{$realm}->{ldap_source}) && ($pf::config::ConfigRealm{$realm}->{ldap_source} eq $ldap) ) {
+            if (defined($pf::config::ConfigRealm{$realm}->{ldap_source_ttls_pap}) && ($pf::config::ConfigRealm{$realm}->{ldap_source_ttls_pap} eq $ldap) ) {
```
Comment (fdurand) [inline lib/pf/services/manager/radiusd_child.pm]: If i am not wrong you should keep the ldap_source too since we use them for "search attributes"

Comment (julsemaan) [reply lib/pf/services/manager/radiusd_child.pm]: Yes, you're right, I missed that

**Comment by julsemaan** [comment]
@fdurand, can you take another look and see if its fine

I think this will have to go in the maintenance too since there were a lot of regressions on EAP TTLS PAP in 11.x
Right now, only Azure AD EAP TTLS PAP is working, any LDAP based TTLS PAP is broken

File: lib/pf/services/manager/radiusd_child.pm [hunk 2]
```diff
@@ -799,18 +802,29 @@ $edir_options
         start = 0
     }
 EOT
+
+        my $client_auth = "";
```
Comment (fdurand) [inline lib/pf/services/manager/radiusd_child.pm:806]: tab

File: lib/pf/services/manager/radiusd_child.pm [hunk 3]
```diff
@@ -799,18 +802,29 @@ $edir_options
...
+	if($ConfigAuthenticationLdap{$ldap}{client_cert_file} && $ConfigAuthenticationLdap{$ldap}{client_key_file}) {
+            $client_auth = <<"EOT";
+       certificate_file = $ConfigAuthenticationLdap{$ldap}{client_cert_file}
+       private_key_file = $ConfigAuthenticationLdap{$ldap}{client_key_file}
+EOT
+	}
```
Comment (fdurand) [inline lib/pf/services/manager/radiusd_child.pm]: tab

File: lib/pf/services/manager/radiusd_child.pm (hunk 2 above)
Comment (julsemaan) [reply lib/pf/services/manager/radiusd_child.pm:806]: replaisaide

File: lib/pf/services/manager/radiusd_child.pm (hunk 3 above)
Comment (julsemaan) [reply lib/pf/services/manager/radiusd_child.pm]: replaisaide

**Comment by fdurand** [comment]
The logic looks to be ok on my side.

**Comment by julsemaan** [comment]
I did extensive tests with my LDAP directory and this seems to hold up and I tested Azure AD as well in another realm and it all works fine

Unless we want this tested by another team member, we'd be ready to merge and get this into the maintenance/11.0 branch

**Comment by fdurand** [comment]
i am ok to merge on my side.

**Pull request merged**